### PR TITLE
[BACKLOG-36879][BACKLOG-37033][BACKLOG-37034] GWT/XUL Dialogs Default, Trap and Restore Focus

### PR DESCRIPTION
**Pull request by dcleao**
Part of PR set: https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1617.

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 47s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtDialog.java [hunk 1]
```diff
@@ -51,6 +52,8 @@ public Element newInstance() {
     } );
   }
 
+  private final static String[] BUTTON_VALUES_AUTO_FOCUS = new String[] { "cancel", "accept", "extra1", "extra2" };
```
Comment (NJtwentyone) [inline gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtDialog.java:55]: Add a comment explaining what these are and/or their function

**Review by NJtwentyone — CHANGES_REQUESTED** [review]
I just have a question about what is variable BUTTON_VALUES_AUTO_FOCUS and how "extra1" is different from "extra2"
@dcleao I assume this build is failing because of other PR dependencies?

File: gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtDialog.java (hunk 1 above)
Comment (dcleao) [reply gwt/impl/src/main/java/org/pentaho/ui/xul/gwt/tags/GwtDialog.java:55]: These are the standard XUL button ids for the `dialog` tag which this class provides an implementation for.
Buttons `extra1` and `extra2` are purposely generic. Their semantic is not defined. As such, there's no way to know which is more _impactful_, so I just listed these two in numeric order.
No constants exist for these identifiers in any of the XUL projects.

In any case, I'll add some more info.

**Comment by buildguy** [comment]
### :x: Build failed in 43s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 49s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by dcleao** [comment]
Merged GWT project. Build is now succeeding.

**Comment by dcleao** [comment]
Rewrote history to remove intermediate commits.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1m 32s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl gwt/impl
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Comment by dcleao** [comment]
@NJtwentyone I addressed your requests. Unfortunately, merging GWT without this one is causing the wcag branch build to fail... As such, I will now merge this one and restart the build. I will address any further requests on a follow up PR.

**Pull request merged**